test/fetchTest.js

```diff
@@ -121,4 +121,28 @@ describe('fetch', function () {
     assert(job.startedon === undefined)
     assert.strictEqual(calledCounter, 2)
   })
+
+  it('should not fetch singleton queue job if there is already an active one', async function () {
+    const boss = this.test.boss = await helper.start(this.test.bossConfig)
+    const queue = this.test.bossConfig.schema
+    const jobOptions = { singletonKey: 'singleton_queue_active_test', useSingletonQueue: true }
+    const sendArgs = [queue, {}, jobOptions]
+
+    const publish1 = await boss.send(...sendArgs)
+    const fetch1 = await boss.fetch(queue)
+    assert(publish1)
+    assert(fetch1)
```

Comment:
One of the 3 travis runs is failing on this line: https://app.travis-ci.com/github/timgit/pg-boss/jobs/597317095

Any idea why this would be intermittent? Could it be due to the `AND startAfter < now()` condition if `now() === startAfter` for a rapid publish+fetch?

Reply:
Ok, figured it out, I was being dumb/sloppy with my correlated subquery qualifiers (yay for CI tests!).

Should be g2g now and I updated the explain analyze plans in the other thread 👍